aws-ssm-parameter/src/main/java/com/amazonaws/ssm/parameter/ReadHandler.java

```diff
@@ -23,20 +32,52 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
             final CallbackContext callbackContext,
             final ProxyClient<SsmClient> proxyClient,
             final Logger logger) {
-        return proxy.initiate("aws-ssm-parameter::resource-read", proxyClient, request.getDesiredResourceState(), callbackContext)
-                .translateToServiceRequest(Translator::getParametersRequest)
-                .makeServiceCall(this::ReadResource)
-                .done((getParametersRequest, getParametersResponse, proxyInvocation, resourceModel, context) -> {
-                    if(getParametersResponse.parameters().size() == 0) {
-                        throw new CfnNotFoundException(ResourceModel.TYPE_NAME, request.getDesiredResourceState().getName());
-                    }
-                    final Parameter parameter = getParametersResponse.parameters().stream().findFirst().get();
-
-                    return ProgressEvent.defaultSuccessHandler(ResourceModel.builder()
-                            .name(parameter.name())
-                            .type(parameter.typeAsString())
-                            .value(parameter.value()).build());
-                });
+        this.logger = logger;
+        // read model only contains primary key
+        ResourceModel model = request.getDesiredResourceState();
+        return ProgressEvent.progress(model, callbackContext)
```

Comment:
I am not sure if this implementation will work. As far as I know, `READ` & `LIST` operations cannot return `IN_PROGRESS` status, they are synchronous.

Reference: https://github.com/aws-cloudformation/cloudformation-cli-java-plugin/blob/ae2e04111756096559256390f4b0d47f6dddac99/src/main/java/software/amazon/cloudformation/AbstractWrapper.java#L335-L337